libr/core/fortune.c

```diff
@@ -1,58 +1,112 @@
-/* radare2 - LGPL - Copyright 2009-2020 - pancake */
+/* radare2 - LGPL - Copyright 2009-2022 - pancake, condret */
 
 #include <r_core.h>
-
-static const char *fortunes[] = {
-	"tips", "fun",
-};
+#include <r_util.h>
 
 static char *getFortuneFile(RCore *core, const char *type) {
+	char *home = r_sys_getenv (R_SYS_HOME);
+	if (!home) {
+		return NULL;
+	}
+	char *path = r_str_newf (R_JOIN_3_PATHS ("%s", R2_HOME_FORTUNES, "fortunes.%s"),
+		home, type);
+	free (home);
+	if (path && r_file_exists (path)) {
+		return path;
+	}
+	free (path);
 	return r_str_newf (R_JOIN_3_PATHS ("%s", R2_FORTUNES, "fortunes.%s"),
 		r_sys_prefix (NULL), type);
 }
 
+R_API RList *r_core_fortune_types(void) {
+	RList *types = r_list_newf (free);
+	if (!types) {
+		return NULL;
+	}
+	char *fortune_dir = r_str_newf (R_JOIN_2_PATHS ("%s", R2_FORTUNES), r_sys_prefix (NULL));
+	if (!fortune_dir) {
+		r_list_free (types);
+	}
+	RList *files = r_sys_dir (fortune_dir);
+	free (fortune_dir);
+	if (!files) {
+		r_list_free (types);
+		return NULL;
+	}
+	RListIter *iter;
+	char *file;
+	r_list_foreach (files, iter, file) {
+		if (r_str_startswith (file, "fortunes.") && file[9]) {
+			r_list_push (types, r_str_new (&file[9]));
+		}
+	}
+	r_list_free (files);
+	fortune_dir = r_str_home (R2_HOME_FORTUNES);
+	if (!fortune_dir) {
+		return types;
+	}
+	files = r_sys_dir (fortune_dir);
+	if (files) {
```

Comment:
move this logic into a separate function so you dont have the sysdir+foreach twice in a function. kind of : `append_types (types, fortune_dir);`